pulsar-functions/runtime-all/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceMain.java

```diff
@@ -101,7 +101,13 @@ public static void main(String[] args) throws Exception {
                 main.getClass().getDeclaredMethod("start", String[].class, ClassLoader.class, ClassLoader.class);
 
         System.out.println("Starting function instance...");
-        method.invoke(main, args, functionInstanceClsLoader, root);
+        try {
+            method.invoke(main, args, functionInstanceClsLoader, root);
+        } catch (InvocationTargetException e) {
+            System.out.println("Failed to start function instance.");
+            e.printStackTrace();
+            System.exit(1);
```

Comment:
System.exit shouldn't be called directly. This makes testing harder. Please take a look how this is handled in the broker.

Reply:
I modeled it after this method: https://github.com/apache/pulsar/blob/ecf83d06082fcdb939ae3faaf8ce3f68c364a557/pulsar-broker/src/main/java/org/apache/pulsar/PulsarBrokerStarter.java#L323-L363

This module has very few dependencies, so I copied the code from the `ShutdownUtil` class.